src/poll/discordEmbedFieldLengthConstraint.ts

```diff
@@ -0,0 +1,37 @@
+import { 
+    ValidatorConstraint, 
+    ValidatorConstraintInterface, 
+    ValidationArguments, 
+    ValidationOptions,
+    registerDecorator
+} from 'class-validator';
+import { ClientConfigDiscordDto } from './poll.dtos'
+
+
+@ValidatorConstraint()
+export class DiscordEmbedFieldLengthConstraint implements ValidatorConstraintInterface {
+  validate(value: any, args: ValidationArguments) {
+    const [relatedPropertyName, lte] = args.constraints;
+    const relatedValue = (args.object as any)[relatedPropertyName] as Array<any>;
+    const discordClientConfig = relatedValue.find((conf) => conf.provider_id === 'discord') as ClientConfigDiscordDto
+    return Array.isArray(value) && typeof Array.isArray(relatedValue) && (value.length + discordClientConfig.role_restrictions.length) <= lte;
+  }
+
+  defaultMessage(args: ValidationArguments) {
+    const [, lte] = args.constraints;
+    return `Sum of poll_options.lenth and role_restrictions.lenth must be less than ${lte + 1}`;
```

Comment:
Help me understand the intent of this?

Reply:
This is a constraint to validate that two variable length arrays don't exceed combined length of `lte`. The discord embed can only have 25 fields max. Each poll option takes a field and each role restriction takes a field. Also the date, vote count, role restrictions headline and strategy take a field each. Lte can be set to the max number of fields that are available for poll options and role restrictions combined.